Monofoxe/Pipefoxe/SpriteGroup/SpriteGroupMathTokenizer.cs

```diff
@@ -0,0 +1,134 @@
+﻿using System.IO;
+using System.Text;
+
+namespace Pipefoxe.SpriteGroup
+{
+  /// <summary>
+  /// This class just goes char by char parsing the text
+  /// works alongside <seealso cref="SpriteGroupMathParser"/> to parse an expresion
+  /// </summary>
+  // based on https://medium.com/@toptensoftware/writing-a-simple-math-expression-engine-in-c-d414de18d4ce
+  public class SpriteGroupMathTokenizer
+  {
+	public enum Token
+	{
+	  Sum = '+',
+	  Sub = '-',
+	  Mul = '*',
+	  Div = '/',
+	  OpenParenthesis = '(',
+	  CloseParenthesis = ')',
+	  Number = 0,
+	  ConstantTop = 1,
+	  ConstantBottom = 2,
+	  ConstantLeft = 3,
+	  ConstantRight = 4,
+	  ConstantCenter = 5,
+	  EOF = -1
+	}
+
+	public Token CurrentToken { get; private set; }
+
+	public int Number { get; private set; }
+
+	TextReader _textReader;
+	char _currentChar;
+	StringBuilder _builder;
+	string _text;
+	public SpriteGroupMathTokenizer(string text)
+	{
+	  _builder = new StringBuilder(text);
+	  _builder.Replace("center", "c").Replace("top", "t").Replace("bottom", "b").Replace("left", "l").Replace("right", "r");
+	  _text = _builder.ToString();
```

Comment:
The string is way too longe, make it something like
```
_builder.Replace("center", "c")
  .Replace("top", "t")
  .Replace("bottom", "b")
```

Reply:
Yuop